web/src/main/java/org/cbioportal/web/StudyController.java

```diff
@@ -68,6 +74,23 @@
         @RequestParam(required = false) StudySortBy sortBy,
         @ApiParam("Direction of the sort")
         @RequestParam(defaultValue = "ASC") Direction direction) {
+        
+        if (keyword == null &&
+            projection == Projection.SUMMARY &&
+            pageSize == 10000000 &&
+            pageNumber == 0 &&
+            sortBy == null &&
+            direction == Direction.ASC
+        ) {
+            synchronized (lock) {
```

Comment:
Have you thought about using a static code block to initialize the defaultResponse and do away with the lock and synchronized block here?

Reply:
Actually, it occurred to me that in the static block the studyService would not yet be initialized.   Using a @PostConstruct annotation would address this - here is an example in [CacheMapUtil.java](https://github.com/cBioPortal/cbioportal/blob/master/persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/util/CacheMapUtil.java#L83)